### ISSUE TITLE
[FEATURE] List available templates when running `dotnet new`

### ISSUE THREAD

**Issue by maxime4000** [nx-dotnet/nx-dotnet#434]
**Describe the bug**
I cannot generate an app from a minimal project. I'm just trying out nx plugins, see how they work and how I could use that to improve the DX, but it seem like I'm not able to use this plugin at all. Missing instruction in the readme ? 

**To Reproduce**
Here the step I have done. 
1. nx create-nx-workspace@latest
2. Choose ts
3. yarn add -D @nrwl/linter @nrwl/angular @nx-dotnet/core
4. nx g @nrwl/linter:workspace-rule eslint-config
5. nx g @nrwl/angular:application angularApp
6. nx g @nx-dotnet/core:app csharpApp
  1. enter x3
7. Voilà

**Expected behavior**
A project with the name csharpApp is create and added to the workspace, but I'm receiving this 

**Screenshots**

```
λ nx g @nx-dotnet/core:app csharpApp
√ What template should the project be initialized with? (https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new#template-options) · 
√ Which language should the project use? · C#
√ Which template should be used for creating the tests project · nunit
Executing Command: dotnet "new" "" "--language" "C#" "--name" "Test2.CsharpApp" "--output" "packages/csharp-app"
System.ArgumentException: Value cannot be null or whitespace. (Parameter 'token')
   at Microsoft.DotNet.Cli.CommandLine.OptionError..ctor(String message, String token, AppliedOption option) in /_/src/source/CommandLine/OptionError.cs:line 21
   at Microsoft.DotNet.Cli.CommandLine.Parser.UnrecognizedArg(String arg) in /_/src/source/CommandLine/Parser.cs:line 170
   at System.Linq.Enumerable.SelectListIterator`2.MoveNext()
   at System.Collections.Generic.List`1.InsertRange(Int32 index, IEnumerable`1 collection)
   at System.Collections.Generic.List`1.AddRange(IEnumerable`1 collection)
   at Microsoft.DotNet.Cli.CommandLine.Parser.Parse(IReadOnlyCollection`1 rawArgs, Boolean isProgressive) in /_/src/source/CommandLine/Parser.cs:line 107
   at Microsoft.DotNet.Cli.CommandLine.Parser.Parse(String[] args) in /_/src/source/CommandLine/Parser.cs:line 31
   at Microsoft.TemplateEngine.Cli.CommandParsing.BaseCommandInput.ParseArgs(String commandName, IReadOnlyList`1 args, Command command)
   at Microsoft.TemplateEngine.Cli.CommandParsing.BaseCommandInput.Parse(String[] args, String commandName)
   at Microsoft.TemplateEngine.Cli.New3Command.ActualRun(String commandName, ITemplateEngineHost host, ITelemetryLogger telemetryLogger, New3Callbacks callbacks, String[] args, String hivePath)
   at Microsoft.TemplateEngine.Cli.New3Command.Run(String commandName, ITemplateEngineHost host, ITelemetryLogger telemetryLogger, New3Callbacks callbacks, String[] args, String hivePath)
   at Microsoft.DotNet.Tools.New.NewCommandShim.Run(String[] args)
   at Microsoft.DotNet.Cli.Program.ProcessArgs(String[] args, TimeSpan startupTime, ITelemetry telemetryClient)
   at Microsoft.DotNet.Cli.Program.Main(String[] args)
dotnet execution returned status code 1
```

**Environment:**

- OS: [e.g. iOS] Windows
- Version [1.9.12](https://www.npmjs.com/package/@nx-dotnet/core/v/1.9.12)

**Comment by maxime4000** [comment]
Ok I'm dumb and I just skip the part where I need to choose a project type. I guess this could be improve! 
This options would be nice :
- Have a list of choice that you can select from. Same type of select as C#, F#, VB, but for the list found on template-options
  - Having Other (unlisted) if you need to select a template that this plugin doesn't know of.

Also, does `npx nx g @nx-dotnet/core:init` does something useful? It seem like it did something, but at the same time, was the correct generate of @nx-dotnet/core:app does the same ? Was that necessary to call init ?

**Comment by AgentEnder** [comment]
Adding some links and notes here for whenever it is implemented.
- Remove x-prompt from schema.json, we will prompt during `normalizeOptions` in project generators.
- Use enquirer + select to run the prompts
- Use `dotnet new --list` to get available template info: https://docs.microsoft.com/en-us/dotnet/core/tools/dotnet-new-list

**Comment by AgentEnder** [comment]
This was released in 12.10, through #436
